src/lib.rs

```diff
@@ -187,38 +214,46 @@ impl Client {
     }
 
     /// Exports metrics by writing to an MMV file
-    pub fn export(&self) -> io::Result<()> {
+    pub fn export(&self, mut metrics: &mut [&mut MMVMetric]) -> io::Result<()> {
```

Comment:
Metrics do need mutable state since the `mmap_view` field is set when `export` is called.

And if we have an `add_metric` method, we'd either have to
- mutably borrow metrics during registration and give their ownership back to the user after export
- make the user pass an `Arc<Metric>` to `add_metric`. That would mean the user always keep `Metric`s inside of `Arc`s (which would impose a runtime cost), or convert them to `Arcs` just for the purpose of `add_metric` and consume it just after. This IMO would add unnecessary complexity to the API.